ui/src/app/components/ct-available-banner-maybe.spec.tsx

```diff
@@ -0,0 +1,124 @@
+
+  const updateProfile = (newTierEligibilities?: UserTierEligibility[], newUserTiers?: string[], newModuleStatus?: AccessModuleStatus) => {
+    const originalProfile = ProfileStubVariables.PROFILE_STUB;
+
+    const unchangedAccessModules: AccessModuleStatus[] =
+      originalProfile.accessModules.modules.filter(m => m.moduleName !== newModuleStatus?.moduleName);
+
+    const profile: Profile = {
+      ...originalProfile,
+      tierEligibilities : newTierEligibilities || originalProfile.tierEligibilities,
+      accessTierShortNames: newUserTiers || originalProfile.accessTierShortNames,
+      accessModules: {
+        modules: [...unchangedAccessModules, newModuleStatus]
+      }
+    }
+
+    profileStore.set({profile, load, reload, updateCache});
+  }
+
+  const updateCdrVersions = (newTier: CdrVersionTier) => {
+    const otherTiers = cdrVersionTiersResponse.tiers.filter(tier => tier.accessTierShortName !== tier.accessTierShortName);
+    cdrVersionStore.set({tiers: [...otherTiers, newTier]});
+  }
+
+  const fulfillAllBannerRequirements = () => {
+    // the user is eligible for the CT
+    const userEligible: UserTierEligibility[] = [
+      {
+        accessTierShortName: AccessTierShortNames.Registered,
+        eraRequired: false,
+        eligible: true
+      },
+      {
+        accessTierShortName: AccessTierShortNames.Controlled,
+        eraRequired: false,
+        eligible: true
+      }
+    ];
+
+    // the user does not currently have CT access
+    const userTiers: string[] = [AccessTierShortNames.Registered];
+
+    // the environment allows users to see the CT (in the UI)
+    environment.accessTiersVisibleToUsers = [AccessTierShortNames.Registered, AccessTierShortNames.Controlled];
+
+    // the user's DUCC access module completion time was before the release of the default CT CDR Version
+    const duccStatus: AccessModuleStatus = {
+      moduleName: AccessModule.DATAUSERCODEOFCONDUCT,
+      completionEpochMillis: 1000
+    }
+    const controlledTierCdrVersions: CdrVersionTier = {
+      ...cdrVersionTiersResponse.tiers.find(t => t.accessTierShortName === AccessTierShortNames.Controlled),
+      defaultCdrVersionCreationTime: 2000
+    }
+
+    // the user is not currently visiting the DAR page
+    window.location.pathname = '/';
+
+    updateProfile(userEligible, userTiers, duccStatus);
+    updateCdrVersions(controlledTierCdrVersions);
+  }
+
+  beforeEach(() => {
+    profileStore.set({profile: ProfileStubVariables.PROFILE_STUB, load, reload, updateCache});
+    cdrVersionStore.set(cdrVersionTiersResponse);
+  });
+
+  it('should render if all of the requirements are met', () => {
+    fulfillAllBannerRequirements();
+
+    const wrapper = component();
+    expect(wrapper.exists()).toBeTruthy();
+
+    expect(ctBannerExists(wrapper)).toBeTruthy();
+  });
+
+  it('should not render if all of the requirements are met, except that the user is not CT eligible', () => {
```

Comment:
Please review this testing approach.  After approval, I'll add tests for the other negative cases.